### PR TITLE
Activity Log: Handle changes to description in API formats

### PR DESCRIPTION
**Pull request by dmsnell**
As we develop a specification for how to describe activity log events we
need to handle any expected transitions in Calypso.

In this PR I have tried to handle all previous and expected future
changes so that Calypso won't be a blocker in the development and
rollout of activity descriptions and titles.

**Testing**

Since the API changes aren't available yet, this isn't too testable other than
to smoke-test in the current **Activity** section under **Stats**. It should
continue to display as **master** displays. As new APIs are testing in your
sandbox this should continue to display properly and add in the activity
titles as the API provides.

### CODE REVIEW

**Comment by dmsnell** [comment]
Thanks @sirreal! Schematified in 1fa6583

File: client/state/data-layer/wpcom/sites/activity/from-api.js [hunk 1]
```diff
@@ -29,6 +29,61 @@ export function transformer( apiResponse ) {
...
 }
 
+/**
+ * Description is an explanation of specific knowledge about an activity
+ *
+ * The description is a formatted explanation of an activity and may
+ * contain links to resources referenced in the explanation
+ *
+ * @param {Object} item from API response
+ * @returns {?Object} parsed notifications-formatted block
```
Comment (sirreal) [inline client/state/data-layer/wpcom/sites/activity/from-api.js]: MINOR: I believe this return type is inaccurate, `{?Object}` is a nullable object, whereas the return here appears to be `{(Object|undefined)}`

File: client/state/data-layer/wpcom/sites/activity/from-api.js [hunk 2]
```diff
@@ -29,6 +29,61 @@ export function transformer( apiResponse ) {
...
+};
+
+/**
+ * Title is a terse and generic summary of the kind of activity
+ *
+ * Usually this includes a noun and a verb acting on that noun
+ * e.g. "Post published" "Theme updated" "Plugin installed" etc…
+ *
+ * @param {Object} item from API response
+ * @returns {?String} activity title
```
Comment (sirreal) [inline client/state/data-layer/wpcom/sites/activity/from-api.js]: MINOR: Same not nullable.

File: client/my-sites/stats/activity-log-item/index.jsx [hunk 3]
```diff
@@ -74,16 +74,14 @@ class ActivityLogItem extends Component {
...
-					<div className="activity-log-item__title">{ activityTitle }</div>
-				) }
 				{ activityDescription && (
 					<div className="activity-log-item__description">
 						{ activityDescription.map( ( part, key ) => (
 							<FormattedBlock key={ key } content={ part } />
 						) ) }
 					</div>
 				) }
+				{ activityTitle && <div className="activity-log-item__title">{ activityTitle }</div> }
```
Comment (sirreal) [inline client/my-sites/stats/activity-log-item/index.jsx]: A bit surprising to see this move down and apparently render alongside `activityDescription`.

Do we expect _either_ `getDescription` or `getTitle` to produce a value from an item, but not both?

File: client/state/data-layer/wpcom/sites/activity/from-api.js (hunk 1 above)
Comment (dmsnell) [reply client/state/data-layer/wpcom/sites/activity/from-api.js]: ooh, good one

Comment (dmsnell) [reply client/state/data-layer/wpcom/sites/activity/from-api.js]: +TypeScript | OCaml

File: client/my-sites/stats/activity-log-item/index.jsx (hunk 3 above)
Comment (dmsnell) [reply client/my-sites/stats/activity-log-item/index.jsx]: this was a response to external changes. we expect to have both a description and title now, and the title goes below

**Pull request merged**